core/txpool/legacypool/legacypool.go

```diff
@@ -1664,14 +1738,16 @@ func (pool *LegacyPool) demoteUnexecutables() {
 		for _, tx := range olds {
 			hash := tx.Hash()
 			pool.all.Remove(hash)
+			pool.removeAuthorities(tx)
```

Comment:
I think we add to the Authorities and remove from the authorities always at the same time when we add/remove from the lookup. I'm wondering whether it makes sense to just track the authorities within the lookup. this way we can kinda hide it a bit from the main structure

I think I will give it a try and see how it looks